Framework/Crystal/src/AnvredCorrection.cpp

```diff
@@ -206,28 +193,46 @@ void AnvredCorrection::exec() {
     // scattered beam
     double scattering = dir.angle(V3D(0.0, 0.0, 1.0));
 
-    Mantid::Kernel::Units::Wavelength wl;
-    std::vector<double> timeflight;
     double depth = 0.2;
+
     double pathlength = 0.0;
+
     std::string bankName;
-    if (m_useScaleFactors)
+
+    if (m_useScaleFactors) {
       scale_init(det, inst, L2, depth, pathlength, bankName);
+    }
+
+    Mantid::Kernel::Units::Wavelength wl;
+    auto points = m_inputWS->points(i);
+
+    // Get a reference to the Y's in the output WS for storing the factors
+    auto &Y = correctionFactors->mutableY(i);
+    auto &E = correctionFactors->mutableE(i);
+
+    // share bin boundaries
+    const auto &inSpec = m_inputWS->getSpectrum(i);
+    correctionFactors->setBinEdges(i, inSpec.binEdges());
```

Comment:
Are you sure this is bin edges? Given the `isHist` flag we had previously, I would assume it is unknown, i.e., we should use `sharedX`?